src/test/java/io/jenkins/plugins/coverage/model/CoveragePluginITest.java

```diff
@@ -0,0 +1,401 @@
+
+        CoverageBuildAction coverageResult = build.getAction(CoverageBuildAction.class);
+        assertThat(coverageResult.getHealthReport().getScore()).isEqualTo(0);
+    }
+
+
+    @Test
+    public void coveragePluginPipelineFailUnhealthyWithResultFailureContainsBUG() {
+        WorkflowJob job = createPipelineWithWorkspaceFiles(JACOCO_FILE_WITH_HIGHER_COVERAGE);
+        job.setDefinition(new CpsFlowDefinition("node {"
+                + "publishCoverage adapters: [jacocoAdapter('**/*.xml')],"
+                + "globalThresholds: [[failUnhealthy: true, thresholdTarget: 'Line', unhealthyThreshold: 96.0, unstableThreshold: 90.0]]"
+                + "}", true));
+
+        Run<?, ?> build = buildWithResult(job, Result.FAILURE);
+        assertThat(build.getNumber()).isEqualTo(1);
+
+        CoverageBuildAction coverageResult = build.getAction(CoverageBuildAction.class);
+        assertThat(coverageResult.getHealthReport().getScore()).isEqualTo(96);
+
+        //HealthReportingAction result = build.getAction(HealthReportingAction.class);
+        //assertThat(result.getBuildHealth()).isEqualTo(null); //healthReport is null why???
+    }
+
+
+    @Test
+    public void coveragePluginPipelineFailUnhealthyWithResultUnstable() {
+        WorkflowJob job = createPipelineWithWorkspaceFiles(JACOCO_FILE_WITH_HIGHER_COVERAGE);
+        job.setDefinition(new CpsFlowDefinition("node {"
+                + "publishCoverage adapters: [jacocoAdapter('**/*.xml')],"
+                + "globalThresholds: [[failUnhealthy: true, thresholdTarget: 'Line', unhealthyThreshold: 90.0, unstableThreshold: 96.0]]"
+                + "}", true));
+
+        Run<?, ?> build = buildWithResult(job, Result.UNSTABLE);
+        assertThat(build.getNumber()).isEqualTo(1);
+    }
+
+
+    @Test
+    public void coveragePluginPipelineFailUnhealthyWithResultSuccess() {
+        WorkflowJob job = createPipelineWithWorkspaceFiles(JACOCO_FILE_WITH_HIGHER_COVERAGE);
+        job.setDefinition(new CpsFlowDefinition("node {"
+                + "publishCoverage adapters: [jacocoAdapter('**/*.xml')],"
+                + "globalThresholds: [[failUnhealthy: true, thresholdTarget: 'Line', unhealthyThreshold: 90.0]]"
+                + "}", true));
+
+        Run<?, ?> build = buildWithResult(job, Result.SUCCESS);
+        assertThat(build.getNumber()).isEqualTo(1);
+    }
+
+
+    @Test
+    public void coveragePluginPipelineFailUnstable() {
+        WorkflowJob job = createPipelineWithWorkspaceFiles(JACOCO_FILE_WITH_HIGHER_COVERAGE);
+        job.setDefinition(new CpsFlowDefinition("node {"
+                + "publishCoverage adapters: [jacocoAdapter('**/*.xml')],"
+                + "failUnstable: true,"
+                + "globalThresholds: [[thresholdTarget: 'Line', unstableThreshold: 96.0]]"
+                + "}", true));
+
+        Run<?, ?> build = buildWithResult(job, Result.FAILURE);
+        assertThat(build.getNumber()).isEqualTo(1);
+    }
```

Comment:
It is hard to understand why it fails in one method only. Maybe adding a comment or improving the method names would help (or using some constants).